codersdk/workspaces.go

```diff
@@ -43,6 +43,30 @@ type Workspace struct {
 	// unlocked by an admin. It is subject to deletion if it breaches
 	// the duration of the locked_ttl field on its template.
 	LockedAt *time.Time `json:"locked_at" format:"date-time"`
+	// Health reports the health of the workspace and its agents.
+	Health WorkspaceHealth `json:"health"`
+}
+
+type WorkspaceHealth struct {
+	Healthy         bool                               `json:"healthy"`          // Healthy is true if the workspace and all of its agents are healthy.
+	FailingSections []string                           `json:"failing_sections"` // FailingSections is a list of sections that have failed their healthcheck.
+	Agents          map[uuid.UUID]WorkspaceAgentHealth `json:"agents"`           // Agents is a map of agent IDs to their health.
```

Comment:
API review: As Go maps don't keep the order, do you think that we should replace it with a corresponding slice?

Reply:
True, maintaining order might be worthwhile, but we'll have to include the agent ID in `WorkspaceAgentHealth` which is duplicated information on the agent endpoints. Perhaps not a big deal though. Failing sections could still work with a slice given that we would use index (e.g. `['agents.0']`).

Reply:
Usually it's annoying when you're refreshing the API and order of items is shuffled every time, so that's the main reason of my recommendation.